Sources/Basic/OutputByteStream.swift

```diff
@@ -405,3 +405,95 @@ public struct Format {
         return TransformedSeparatedListStreamable(items: items, transform: transform, separator: separator)
     }
 }
+
+/// Represents a stream which is backed to a file. Not for instantiating.
+public class FileOutputByteStream: OutputByteStream {
+
+    /// Closes the file flushing any buffered data.
+    public func close() {
+        // For subclasses to implement.
+    }
+}
+
+import libc
+
+/// Implements file output stream for local file system.
+public final class LocalFileOutputByteStream: FileOutputByteStream {
+    
+    /// The pointer to the file.
+    private let fp: UnsafeMutablePointer<FILE>
+
+    /// Instantiate using the file pointer.
+    init(fp: UnsafeMutablePointer<FILE>) throws {
+        self.fp = fp
+        super.init()
+    }
+
+    /// Opens the file for writing at the provided path.
+    public init(_ path: AbsolutePath) throws {
+        guard let fp = fopen(path.asString, "wb") else {
+            throw FileSystemError(errno: errno)
+        }
+        self.fp = fp
+        super.init()
+    }
+    
+    override public func flush() throws {
+        var contents = buffer
+        while true {
+            let n = fwrite(&contents, 1, contents.count, fp)
+            if n < 0 {
+                if errno == EINTR { continue }
+                throw FileSystemError.ioError
+            }
+            if n != contents.count {
```

Comment:
I thought that `fwrite()` could return a count less than the number you passed in as the parameter, in which case you should write the remaining items in a loop until you have written all or until an error (other than `EINTR`) occurs.  That's definitely the case for `write()` — perhaps `fwrite()` does it's own loop, do you know if that's the case?